apps/content-gateway-api/src/app/repository/MongoUserRepository.spec.ts

```diff
@@ -0,0 +1,120 @@
+// eslint-disable-next-line @nrwl/nx/enforce-module-boundaries
+import { extractRight, programError } from "@banklessdao/util-misc";
+import { UserNotFoundError, UserRepository } from "@domain/feature-gateway";
+import * as E from "fp-ts/Either";
+import { Collection, Db, MongoClient } from "mongodb";
+import { v4 as uuid } from "uuid";
+import { createMongoUserRepository } from ".";
+import { MongoUser } from "./mongo/MongoUser";
+
+const url =
+    process.env.MONGO_CGA_URL ?? programError("MONGO_CGA_URL is missing");
+const dbName =
+    process.env.MONGO_CGA_USER ?? programError("MONGO_CGA_USER is missing");
+
+describe("Given a Mongo user repository", () => {
+    let target: UserRepository;
+    let mongoClient: MongoClient;
+    let db: Db;
+    let users: Collection<MongoUser>;
+
+    const collName = uuid();
+
+    beforeAll(async () => {
+        mongoClient = new MongoClient(url);
+        db = mongoClient.db(dbName);
+        users = db.collection<MongoUser>(collName);
+        await mongoClient.connect();
+
+        target = await createMongoUserRepository({
+            db,
+            collName,
+        });
+    });
+
+    afterAll(async () => {
+        await users.drop();
+        await mongoClient.close();
+    });
+
+    describe("When creating a new user", () => {
+        it("Then it is successfully created when valid", async () => {
```

Comment:
It would be if there was validation, but currently it is not possible to call this method and get an error based on the input. 😅

Reply:
Would it make sense to restrict the amount of users someone can create? Just in case someone intentionally floods our db with new users, or some programming error from a client creates millions of users